### DISCUSSION TITLE
How to handle year and date in template?

### DISCUSSION THREAD

**Discussion by wztdream** [org-roam/org-roam-bibtex#254] [Q&A] [answered]
Hi,

Some bibtex entry only have date field others have year filed, I set a template to escape "year" as a org headline, but it will be empty if the bibtex entry only have "date" field. Is there a field keyword that can handle both conditions:
1. bibtex entry have year field -> return year
2. bibtex entry have date field -> convert it to year

**Comment by myshevchuk** [comment, marked as answer]
Hi,

no, there is no such keyword. You are welcome to submit a feature request. It will take some time though because I'll be quite busy until the end of next week.

You can achieve what you want with LISP expressions in your template. Read the documentation of `org-capture-templates`.
You can have a function, for example, which takes both the keywords and then returns a non-empty result if one of the fields is non-empty. It'd look something like this:

```elisp
(defun my-orb-year-or-date (keyword1 keyword2)
  (if (string-empty-p keyword1)
    keyword2
  keyword1))
```

Adjust it to your needs to handle the conversion of date to year. And then in the templates:

```elisp
(setq org-roam-capture-templates
'(("r" "reference" plain ""
  :target
  (file+head "${citekey}.org" "#+TITLE: ${title}
  %(my-orb-year-or-date \"%^{year}\" \"%^{date}\")"))))
```

**Reply by wztdream** [reply]
thank you, your workaround works for me! Also I add a feature request https://github.com/org-roam/org-roam-bibtex/issues/255
